### ISSUE TITLE
Shape.from_records should accept pandas DataFrame

### ISSUE THREAD

**Issue by mangecoeur** [ioam/geoviews#41]
When trying to use `from_records` to merge shape records with a dataframe I get:

```
/Users/<>/miniconda3/envs/science/lib/python3.5/site-packages/geoviews/element/geo.py in from_records(cls, records, dataset, on, value, index, **kwargs)
    316         Returns an NdOverlay of Shapes.
    317         """
--> 318         if dataset and not on:
    319             raise ValueError('To merge dataset with shapes mapping '
    320                              'must define attribute(s) to merge on.')

/Users/<>/miniconda3/envs/science/lib/python3.5/site-packages/pandas/core/generic.py in __nonzero__(self)
    915         raise ValueError("The truth value of a {0} is ambiguous. "
    916                          "Use a.empty, a.bool(), a.item(), a.any() or a.all()."
--> 917                          .format(self.__class__.__name__))
    918 
    919     __bool__ = __nonzero__

ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
```

**Comment by philippjfr** [comment]
The ``from_records`` method expects a HoloViews type. You can simply wrap the dataframe in an ``hv.Dataset``, see the example in the docs for more detail:

```python
referendum = pd.read_csv('./assets/referendum.csv')
referendum = hv.Dataset(referendum)
gv.Shape.from_records(shapes.records(), referendum, on='code', value='leaveVoteshare',
                     index=['name', 'regionName'], crs=ccrs.PlateCarree())
```

**Comment by mangecoeur** [comment]
Thanks, I missed the bit in the example where the dataframe was wrapped in a dataset. Perhaps this could be done automatically for pandas dataframes?

**Comment by philippjfr** [comment]
>Thanks, I missed the bit in the example where the dataframe was wrapped in a dataset. Perhaps this could be done automatically for pandas dataframes?

Yes, not a bad idea.

**Comment by philippjfr** [comment]
Can't guarantee I'll get to it myself in the near future, very happy to accept PRs though.

**Comment by philippjfr** [comment]
This was implemented in be9eb7b031bccd5517887b4db9de3a771cb01584. Closing.
